src/pages/api/fetchSummary.ts

```diff
@@ -0,0 +1,62 @@
+import { Configuration, OpenAIApi } from 'openai';
+import { mockData, DataType } from '../../../mock-response';
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { generatePrompt } from '~/utlis/generatePrompt';
+
+const configuration = new Configuration({
+  apiKey: process.env.OPENAI_API_KEY,
+});
+const openai = new OpenAIApi(configuration);
+
+type Error = {
+  error: { message: string };
+};
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<DataType | Error>,
+) {
+  if (!configuration.apiKey) {
+    res.status(500).json({
+      error: {
+        message:
+          'OpenAI API key not configured, please follow instructions in README.md',
+      },
+    });
+    return;
+  }
+
+  const { query } = req;
+  const article = query.article || '';
```

Comment:
Once my PR is merged, if we're still using query parameters you can do something like this:

```suggestion
const article = getStringOrFirst(query.article)
```

This ensures we're getting a `String` response from the query parameters as occasionally these can return arrays which would conflict with what typescript is expecting :)